e2e/test/scenarios/permissions/create-queries/granular.cy.spec.js

```diff
@@ -0,0 +1,96 @@
+import { USER_GROUPS } from "e2e/support/cypress_data";
+import {
+  assertPermissionTable,
+  selectSidebarItem,
+  popover,
+  selectPermissionRow,
+  modal,
+  restore,
+  modifyPermission,
+} from "e2e/support/helpers";
+
+const { ALL_USERS_GROUP } = USER_GROUPS;
+
+const NATIVE_QUERIES_PERMISSION_INDEX = 1;
+
+describe("scenarios > admin > permissions > create queries > granular", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should allow configuring granular permissions in group view", () => {
+    cy.visit(`/admin/permissions/data/group/${ALL_USERS_GROUP}`);
+
+    modifyPermission(
+      "Sample Database",
+      NATIVE_QUERIES_PERMISSION_INDEX,
+      "Query builder and native",
+    );
+
+    // should allow choosing granular option at the db level
+    modifyPermission(
+      "Sample Database",
+      NATIVE_QUERIES_PERMISSION_INDEX,
+      "Granular",
+    );
+
+    assertPermissionTable([
+      ["Accounts", "Can view", "Query builder and native"],
+      ["Analytic Events", "Can view", "Query builder and native"],
+      ["Feedback", "Can view", "Query builder and native"],
+      ["Invoices", "Can view", "Query builder and native"],
+      ["Orders", "Can view", "Query builder and native"],
+      ["People", "Can view", "Query builder and native"],
+      ["Products", "Can view", "Query builder and native"],
+      ["Reviews", "Can view", "Query builder and native"],
```

Comment:
Would it be less jarring if we automatically change tables to `Query builder` if you click into `Granular`? Since if you're setting granular settings, they'll have to change anyway?

Just an idea, don't need to change necessarily.

Reply:
I had actually implemented it that way initially, but ended up changing it to the way it is now. My thinking was that without an undo button for the last change it's kind of annoying to have a change made for you if you were just exploring the idea of schema/table level permissions but don't end up doing it. In the case where you've made other changes somewhere else, you couldn't just hit cancel, but would have to go back up to the db level to change the value back to query builder and native. Also, yes it's jarring, but I felt like it was a bit clearer seeing the change happen right before your eyes rather than landing on the view with the change having already been made.

Could totally be convinced to do it differently, but I do think both approaches benefit from being warned something is about to take place. Hoping to get that in tomorrow.

Reply:
That makes sense